### PR TITLE
3645 add item name column to requisition line

### LINKED ISSUE #3645: Add `item_name` column to requisition line
## Is your feature request related to a problem? Please describe 👀
When a item has been merged any requisition lines should retain the deleted item name for displaying purposes. 

## Describe the solution you'd like 🎁
Add `item_name` column to requisition line and populate where needed

### Moneyworks Jobcode 🧰
OMS:MERGE

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #3645 

# 👩🏻‍💻 What does this PR do?
Adds an `item_name` column to `requisition_line` so that the original item name can be displayed if an item has been merged. Also if an item has been selected then the item name in the edit modal will show the original item name.

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Have two items
- [ ] Create an `Internal Order` with one of the items
- [ ] Go to OG and merge the two items, keeping the one that isn't in the Internal Order
- [ ] Sync
- [ ] Go back to the Internal Order - should see that the item name is the original item name not the merged one

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx [hunk 1]
```diff
@@ -112,25 +113,38 @@ export const RequestLineEditForm = ({
...
-          }
-          extraFilter={
-            disabled
-              ? undefined
-              : itemRow => !lines?.some(({ item }) => itemRow.id === item.id)
-          }
-        />
+        <>
+          {(disabled && (
+            <BasicTextInput
```
Comment (roxy-dao) [inline client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:124]: Using a disabled text input so it at least looks somewhat like the autocomplete component kinda... 👀

File: server/repository/src/migrations/v1_07_01/stock_on_hand_add_item_name.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    sql!(
+        connection,
+        r#"
+        DROP VIEW stock_on_hand;
+
+        CREATE VIEW stock_on_hand AS
+        SELECT
+          'n/a' AS id,
+          items_and_stores.item_id AS item_id,
+          items_and_stores.item_name AS item_name,
```
Comment (roxy-dao) [inline server/repository/src/migrations/v1_07_01/stock_on_hand_add_item_name.rs:19]: The requisition lines are generated by item_stats so had to modify the stock_on_hand view

File: server/service/src/sync/translations/requisition_line.rs [hunk 3]
```diff
@@ -152,14 +154,6 @@ impl SyncTranslation for RequisitionLineTranslation {
             ))?
             .item_id;
 
-        // Required for backward compatibility (authorisation web app uses this to display item name)
-        let item_name = ItemRowRepository::new(connection)
```
Comment (roxy-dao) [inline server/service/src/sync/translations/requisition_line.rs:156]: Can be removed since it's being stored in the table now

**Comment by mark-prins** [comment]
Just checking - should this be targeting `main`? it doesn't have a hotfix label, but has 'enhancement' 🤔

File: client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx [hunk 4]
```diff
@@ -114,25 +115,38 @@ export const RequestLineEditForm = ({
   const formatNumber = useFormatNumber();
   const { lines } = useRequest.line.list();
   const requestedQuantity = draftLine?.requestedQuantity ?? 0;
+  const originalItemName = lines?.find(({ item }) => item.id === item?.id)
```
Comment (clemens-msupply) [inline client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:118]: This doesn't look right: `item.id === item.id`

File: client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx [hunk 5]
```diff
@@ -114,25 +115,38 @@ export const RequestLineEditForm = ({
...
-          extraFilter={
-            disabled
-              ? undefined
-              : itemRow => !lines?.some(({ item }) => itemRow.id === item.id)
-          }
-        />
+        <>
+          {(disabled && (
+            <BasicTextInput
+              value={`${item?.code}     ${originalItemName}`}
```
Comment (clemens-msupply) [inline client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:127]: Can't the item.code change in a merge?

File: client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx [hunk 6]
```diff
@@ -114,25 +115,38 @@ export const RequestLineEditForm = ({
   const formatNumber = useFormatNumber();
   const { lines } = useRequest.line.list();
   const requestedQuantity = draftLine?.requestedQuantity ?? 0;
+  const originalItemName = lines?.find(({ item }) => item.id === item.id)
+    ?.itemName;
```
Comment (clemens-msupply) [inline client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:119]: sorry, I meant `item.id === item.id` doesn't make sense to me its always true.

How is this working in general? were is the item in the props coming from? and why are you using it to find a line from within the RequestLineEditForm? Shouldn't there already be a itemName in the current line?

**Review by clemens-msupply — CHANGES_REQUESTED** [review]

File: client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx (hunk 5 above)
Comment (roxy-dao) [reply client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:127]: Yeah, but I was told only the item name should stay the same

File: client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx (hunk 6 above)
Comment (roxy-dao) [reply client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:119]: Ah I was trying to get the line item and prop item at the same time but can see how that is confusing and might fudge the code... prop item is the selected item

Comment (roxy-dao) [reply client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:119]: Item from prop is the currently selected item and so I'm trying to get the item name from the line

File: client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx (hunk 5 above)
Comment (clemens-msupply) [reply client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:127]: Ah yes, thanks remember now

File: client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx (hunk 6 above)
Comment (roxy-dao) [reply client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:119]: I can rename to currentItem ~

**Review by clemens-msupply — APPROVED** [review]
Thanks looks good to me

**Pull request merged**